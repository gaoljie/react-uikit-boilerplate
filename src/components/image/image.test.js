import React from "react";
import "jest-styled-components";
import { shallow, mount } from "enzyme";

import Image from "./index";

const prop = {
  src: "url.com",
  w: 180,
  h: 110,
  of: "cover",
  width: "200px",
  height: "200px",
  alt: "image"
};

describe("Test Image", () => {
  it("Matches the snapshot", () => {
    const wrapper = shallow(<Image {...prop} />);
    expect(wrapper).toMatchSnapshot();
  });

  it("Render Prop", () => {
    const wrapper = mount(<Image {...prop} lazyLoad={false} />);
    const img = wrapper.find("img");
    expect(img.props().src.includes(`w_${prop.w}`)).toBe(true);
    expect(img.props().src.includes(`h_${prop.h}`)).toBe(true);
    expect(img.props().src.includes(`c_`)).toBe(false);

    wrapper.setProps({ w: null, h: 120, mode: "crop" });
    expect(
      wrapper
        .find("img")
        .props()
        .src.includes(`w_`)
    ).toBe(false);
    expect(
      wrapper
        .find("img")
        .props()
        .src.includes(`h_120`)
    ).toBe(true);

    expect(
      wrapper
        .find("img")
        .props()
        .src.includes(`c_crop`)
    ).toBe(true);

    wrapper.setProps({ w: 120, h: null });
    expect(
      wrapper
        .find("img")
        .props()
        .src.includes(`w_120`)
    ).toBe(true);
    expect(
      wrapper
        .find("img")
        .props()
        .src.includes(`h_`)
    ).toBe(false);
    expect(wrapper.find("img").props().alt).toEqual(prop.alt);

    wrapper.setProps({
      w: null,
      h: null,
      src: "aa.gif",
      xl: 1,
      lg: null,
      md: null,
      sm: null
    });
    expect(
      wrapper
        .find("img")
        .props()
        .src.includes(`w_`)
    ).toBe(false);
    expect(
      wrapper
        .find("img")
        .props()
        .src.includes(`h_`)
    ).toBe(false);

    expect(
      wrapper
        .find("img")
        .props()
        .src.includes(`fl_lossy/`)
    ).toBe(true);

    expect(wrapper).toHaveStyleRule("object-fit", prop.of);

    expect(wrapper).toHaveStyleRule("width", "100%");

    wrapper.setProps({ xl: null, lg: 1, md: null, sm: null });

    expect(wrapper).toHaveStyleRule("width", "100%");

    wrapper.setProps({ xl: null, lg: null, md: 1, sm: null });

    expect(wrapper).toHaveStyleRule("width", "100%");

    wrapper.setProps({ xl: null, lg: null, md: null, sm: 1 });

    expect(wrapper).toHaveStyleRule("width", "100%");

    wrapper.setProps({ overflow: true, offset: 300, lazyload: true });
  });
});
