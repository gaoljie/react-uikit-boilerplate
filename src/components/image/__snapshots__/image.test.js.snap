// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Image Matches the snapshot 1`] = `
<LazyLoad
  offset={200}
  once={false}
  overflow={false}
  placeholder={
    <ForwardRef(styled.div)
      alt="image"
      h={110}
      height="200px"
      of="cover"
      src="url.com"
      w={180}
      width="200px"
    />
  }
  resize={false}
  scroll={true}
  unmountIfInvisible={false}
>
  <Styled(styled.div)
    alt="image"
    as="img"
    h={110}
    height="200px"
    of="cover"
    src="https://res.cloudinary.com/helloyumi/image/upload/q_70/w_180,h_110/s3/url.com"
    w={180}
    width="200px"
  />
</LazyLoad>
`;
